File: Ros/ROS-Trials/c++/optitracklistener.cpp
#include "ros/ros.h"
#include "geometry_msgs/PoseStamped.h"
#include "geometry_msgs/Pose.h"

void callback(const geometry_msgs::PoseStamped::ConstPtr& msg)
{
  ROS_INFO("%f  %f  %f", msg->pose.position.x, msg->pose.position.y, msg->pose.position.z);
  ROS_INFO("%f  %f  %f  %f", msg->pose.orientation.x, msg->pose.orientation.y, msg->pose.orientation.z, msg->pose.orientation.w);
}

int main(int argc, char **argv) 
{
  ros::init(argc, argv, "optitracklistener");
  ros::NodeHandle n;
  ros::Subscriber sub = n.subscribe("geometry_msgs/Pose", 1000, callback);
  ros::spin();

  return 0;
}

File: Ros/ROS-Trials/simple-optitracklist-python-cpp/python/optitracklistener.py
#!/usr/bin/env python3
import rospy
from geometry_msgs.msg import PoseStamped

def callback(data):
  #rospy.loginfo(rospy.get_caller_id() + "I heard %s", data.data)
  #print(data)

  print('----------------------------------')
  print(data.header.stamp)
  print(data.pose.position.x)
  print(data.pose.position.y)
  print(data.pose.position.z)
  print(data.pose.orientation.x)
  print(data.pose.orientation.y)
  print(data.pose.orientation.z)
  print(data.pose.orientation.w)


def listener():
  rospy.init_node('optitracklistener', anonymous=True)
  rospy.Subscriber('geometry_msgs/Pose', PoseStamped, callback)
  rospy.spin()

if __name__ == '__main__':
  listener()
